lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -211,11 +212,45 @@ def find_matches(self, requirements):
         first_req = requirements[0]
         fqcn = first_req.fqcn
         # The fqcn is guaranteed to be the same
-        coll_versions = self._api_proxy.get_collection_versions(first_req)
+        version_req = "A SemVer-compliant version or '*' is required. See https://semver.org to learn how to compose it correctly. "
+        version_req += "This is an issue with the collection."
+        try:
+            coll_versions = self._api_proxy.get_collection_versions(first_req)
+        except TypeError as exc:
+            if first_req.is_concrete_artifact:
+                # Non hashable versions will cause a TypeError
+                raise ValueError(
+                    f"Invalid version found for the collection '{first_req}'. " + version_req
+                ) from exc
+            # Unexpected error from a Galaxy server
+            raise
+
         if first_req.is_concrete_artifact:
             # FIXME: do we assume that all the following artifacts are also concrete?
             # FIXME: does using fqcn==None cause us problems here?
 
+            # Ensure the version found in the concrete artifact is SemVer-compliant
+            for version, req_src in coll_versions:
+                version_err = f"Invalid version found for the collection '{first_req}': {version} ({type(version)}). " + version_req
```

Comment:
Same as above:
```suggestion
                version_err = f"Invalid version found for the collection '{first_req}': {version} ({type(version)}). {version_req}"
```
(unless there's a good reason for doing f-string+concat that I'm missing?)